frontend/src/metabase/query_builder/components/dataref/ModelPane/ModelPane.tsx

```diff
@@ -0,0 +1,79 @@
+import React from "react";
+import { connect } from "react-redux";
+import { t, jt } from "ttag";
+import PropTypes from "prop-types";
+
+import DateTime from "metabase/components/DateTime";
+import {
+  Description,
+  EmptyDescription,
+} from "metabase/components/MetadataInfo/MetadataInfo.styled";
+import { getQuestionFromCard } from "metabase/query_builder/selectors";
+import type { Card } from "metabase-types/api";
+import type { State } from "metabase-types/store";
+import Question from "metabase-lib/lib/Question";
+import FieldList from "../FieldList";
+import {
+  ModelPaneDetail,
+  ModelPaneDetailLink,
+  ModelPaneDetailLinkText,
+  ModelPaneDetailText,
+  ModelPaneIcon,
+  ModelPaneDescription,
+} from "./ModelPane.styled";
+
+interface ModelPaneProps {
+  show: (type: string, item: unknown) => void;
+  model: Card;
+  question: Question;
+}
+
+const mapStateToProps = (state: State, props: ModelPaneProps) => ({
+  question: getQuestionFromCard(state, props.model),
```

Comment:
@ranquild I've changed this file to typescript, but this getQuestionFromCard function call doesn't typecheck. It's defined here, in this jsx file. It works fine at runtime. How should I resolve it?
https://github.com/metabase/metabase/blob/cb755580a6e65f7d0bd0cc7b8451617266fda20d/frontend/src/metabase/query_builder/selectors.js#L333

Reply:
Maybe a JSDoc annotation would work here (at this constant declaration)? https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html#type

Reply:
I tried adding this type declaration, but since it's a comment I get import unused errors with `State` and `SavedCard`. Should we layer another hack on top to trick the linter into thinking we've used them?
```
/** @type {function(State, SavedCard): Question} */
```

Reply:
Using `unknown` fixes it, as another workaround
https://github.com/metabase/metabase/pull/25494/commits/3bf5a0470b3aeb7d9eb0b8069b2c069a1c382020